src/buildApplication/buildMetadataFiles.ts

```diff
@@ -1,27 +1,41 @@
 import { join } from 'path';
 import { writeFile } from 'fs/promises';
 import { nextOnPagesVersion } from '../utils';
+import { getPhaseRoutes, getVercelConfig } from './getVercelConfig';
 
 /**
  * Builds metadata files needed for the worker to correctly run.
  */
 export async function buildMetadataFiles() {
-	await Promise.all([buildHeaders(), buildRoutes()]);
+	await Promise.all([buildNextStaticHeaders(), buildRoutes()]);
 }
 
-async function buildHeaders() {
-	await writeFile(
-		join('.vercel', 'output', 'static', '_headers'),
-		`
+async function buildNextStaticHeaders() {
+	const vercelConfig = await getVercelConfig();
+
+	const hitRoutes = getPhaseRoutes(vercelConfig, 'hit');
+
+	const nextStaticRoute = hitRoutes.find(route =>
+		route.src?.startsWith('/_next/static/')
+	);
+	const nextStaticHeaders = (nextStaticRoute as VercelSource).headers;
```

Comment:
`nextStaticRoute` could be null here, right?

Or is it guaranteed to find one?

Reply:
Perhaps this should be 

```suggestion
	const nextStaticHeaders = (nextStaticRoute as VercelSource)?.headers;
```

Reply:
I think it is always guaranteed to find one 🤔

but I don't mind having more defensive code here to make sure things are as they should be 🙂👍 

(it's been a while since I opened this PR I am not sure if I missed the potential `undefined` because of the casting or if I purposely omitted it since we should always find the headers 🤔)